linuxPackages.nvidia_x11.persistenced: fix build

phanirithvij opened the pull request:
fixes #437066

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

phanirithvij committed:
linuxPackages.nvidia_x11.persistenced: fix build

Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>

phanirithvij commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 439514`

Logs: https://github.com/phanirithvij/nixpkgs-review-gha/actions/runs/17402278898

gshpychka commented on pkgs/os-specific/linux/nvidia-x11/persistenced.nix:
```diff
@@ -39,10 +54,10 @@ stdenv.mkDerivation {
-  meta = with lib; {
+  meta = {
     homepage = "https://www.nvidia.com/object/unix.html";
     description = "Settings application for NVIDIA graphics cards";
-    license = licenses.unfreeRedistributable;
+    license = lib.licenses.mit;
```
is it really MIT?

ilya-epifanov commented:
running fine so far (tested for a couple of sleep cycles)

JohnRTitor commented on pkgs/os-specific/linux/nvidia-x11/persistenced.nix:
```diff
@@ -19,8 +21,21 @@ stdenv.mkDerivation {
-  buildInputs = [ libtirpc ];
+  configurePhase = ''
+    runHook preConfigure
+    export LIBRARY_PATH="${glibc}/lib"
```
Can we not use `env` for this?

JohnRTitor commented on pkgs/os-specific/linux/nvidia-x11/persistenced.nix:
```diff
@@ -39,10 +50,10 @@ stdenv.mkDerivation {
   env.NIX_CFLAGS_COMPILE = toString [ "-I${libtirpc.dev}/include/tirpc" ];
   NIX_LDFLAGS = [ "-ltirpc" ];
```
You might want to club the envs together at the top.

phanirithvij committed:
linuxPackages.nvidia_x11.persistenced: update metadata

license is mit as per source repo
update homepage url to github source
remove with lib usage in meta

Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>
